### ISSUE TITLE
ask for checkpoints

### ISSUE THREAD

**Issue by zhouak47** [LJY-XCX/RFTrans#2]
Thanks for your great job. I have tried to train F2NET as you told in readme. But the result is not good.  I don't know how to solve this problem.So could you please provide the checkpoints? thanks

**Comment by LJY-XCX** [comment]
Thank you for your interest for RFTrans!
We don't have the pretrained checkpoints currently. However, I am ready to help you debug. Could you share some details about your training process? Did you use the generated refractive flow in the training dataset?

**Comment by zhouak47** [comment]
Thank you so much for your kind reply. I use the RGB pictures and generated refractive flow in the dataset you provide.  The refractive flow predicted by the F2NET is as follows.

![Image](https://github.com/user-attachments/assets/2207eab5-dcd5-482a-9a4b-1f8790a1a753)
the flow in the dataset is as below

![Image](https://github.com/user-attachments/assets/081061b8-360c-4828-8711-e0e767204968)

Is this  predicted refractive flow correct? it seems to have a big difference from the flow in the dataset,my config file is as below.The training loss is around 390, and the val loss is around 32.Do i need to train more epochs? Could you help me? I look forward to your reply.



# train.py Config - Training
train:
  # For datasets, please pass atleast 1 value. If no datasets exist, pass "" as path for images.
  # Synthetic datasets with ground truth labels
  datasetsTrain:
  - images: '/home/xxw/dataset/train/RGB'
    labels: '/home/xxw/dataset/train/flow'
    masks: '/home/xxw/dataset/train/mask'

  # Synthetic datasets with ground truth labels - 10% split of train
  datasetsVal:
    - images: '/home/xxw/dataset/train/RGB'
      labels: '/home/xxw/dataset/train/flow'
      masks: '/home/xxw/dataset/train/mask'

  # Real Images (no ground truth labels)
  datasetsTestReal:

  # Synthetic datasets with ground truth labels - Used as test set
  datasetsTestSynthetic:

  # Training/Validation Params
  model: "drn" # Possible values: ['deeplab_xception', 'deeplab_resnet', 'drn']
  batchSize: 12
  batchSizeMatterport: 0
  batchSizeScannet: 0
  validationBatchSize: 32
  testBatchSize: 16
  numEpochs: 401
  imgHeight: 512
  imgWidth: 512
  numClasses: 2
  numInputChannels: 3 # Num of channels in input image. RGB = 3 channels, Grayscale = 1 channel.
  numWorkers: 4 # Num of workers used in the dataloader
  logsDir: "/home/xxw/RFTrans/results" # Directory where logs of each exp will be saved.
  #lossFunc: "cosine" # Possible values: ['cosine', 'radians']
  percentageDataForTraining: 1.0 # The percentage of images in dataset to be used for training.
  percentageDataForMatterportTraining: 0.5 # The percentage of images in dataset to be used for training.
  percentageDataForScannetTraining: 0.35
  percentageDataForValidation: 1.0
  percentageDataForMatterportVal: 0.5
  percentageDataForScannettVal: 0.5

  # Deeplab specific
  outputStride: 8 # Possible values: [8, 16]. Output stride for deeplabv3 model. Smaller values give finer details in output mask.
  epochSize: 1 # After these many epochs, change learning rate

  continueTraining: true # If true, continue training from a checkpoint
  pathPrevCheckpoint: "/home/xxw/RFTrans/results/exp-017/checkpoints/checkpoint-epoch-0290.pth" # Path to .pth checkpoint file to load to continue training from
  initOptimizerFromCheckpoint: True # Re-Initialize optimizer's state from checkpoint. NOTE: when this is enabled, value of learningRate will be overridden with value from checkpoint.
  loadEpochNumberFromCheckpoint: True # If true, the epoch/iter numbering will start from the checkpoint's last epoch num.

  saveImageInterval: 1 # Log output images to tensorboard every saveImageInterval epochs
  saveImageIntervalIter: 1000 # Every N iterations, log output images to tensorboard
  testInterval: 10 # Run on test set every nTestInterval epochs. Keep at 0 to skip tests.
  saveModelInterval: 10 # Save the model checkpoints every N epochs

  # Optimizer Params
  optimizer: "Adam" # Possible Values: ["SGD", "Adam"]
  optimAdam:
    learningRate: 5e-5  # Origin value: 0.0001
    weightDecay: 0 # Other values: 0.0001
  optimSgd:
    learningRate: 1e-6
    momentum: 0.9
    weight_decay: 5e-4
  lrScheduler: "StepLR" # Possible Values: ['', 'StepLR', 'ReduceLROnPlateau'] 
  lrSchedulerStep:
    step_size: 1000
    gamma: 0.1
  lrSchedulerPlateau:
    factor: 0.8
    patience: 25
    verbose: True

# eval.py Config - Validation/Testing Inference
eval:
  # Synthetic datasets with ground truth labels
  # Used as validation set
  datasetsSynthetic:
    - images: '/home/xxw/dataset/train_syn/RGB'
      labels: '/home/xxw/dataset/train_syn/flow'
      masks: '/home/xxw/dataset/train_syn/mask'

  # Datasets of real images, no labels available
  # Used as Test set
  datasetsReal:


  # Params
  model: "drn" # Possible values: ['deeplab_xception', 'deeplab_resnet', 'drn']
  numClasses: 2
  batchSize: 64
  imgHeight: 512
  imgWidth: 512
  os: 8
  numWorkers: 4 # Num of workers used in the dataloader

  pathWeightsFile: "/home/xxw/RFTrans/results/exp-017/checkpoints/checkpoint-epoch-0400.pth" # Path to the checkpoint to be loaded
  resultsDir: "test_result"

**Comment by LJY-XCX** [comment]
First, for refractive flow prediction, RFNet should be used instead of F2Net. Also, are you using the updated training set from HuggingFace? Seems like these transparent object models in the provided image are from ClearGrasp, which we used for ablation.

**Comment by zhouak47** [comment]
Thanks for your kind reply. I made a mistake when response. Actually i used RFnet for refractive prediction. And i also used training set to train RFNet.THE epoch number is 590,and the result is below .Is this result correct?  

![Image](https://github.com/user-attachments/assets/580ab99c-f2e5-4e0d-86db-95e7eefb8b26)

**Comment by LJY-XCX** [comment]
This result doesn't look correct to me. I notice that you changed the default batchSize but didn't change the corresponding learning rate. This might be a possible issue.

**Comment by zhouak47** [comment]
Thank you so much for your kind reply. I will try to use the default batchsize and train again.Hope this result could be correct.

**Comment by zhouak47** [comment]
I have trained another network using the default batchSize and other parameters , but the result  below seems to be incorrect. What should i do now? Do i need to train RFnet and F2Net together？I am looking forward for your help.

![Image](https://github.com/user-attachments/assets/d2c28b09-a3f3-4507-8fcc-16fc8273daa9)

**Comment by LJY-XCX** [comment]
Can you check the training curve with tensorboard? Also, how does this model perform in validation set (which has the same distribution of training set)? 
If needed, we can set up a zoom meeting to discuss about the results.

**Comment by zhouak47** [comment]
Thanks a lot for your kind reply.The training curve and pictures are as below.Do i need to train more steps , because it seems that the network has not been trained well.

![Image](https://github.com/user-attachments/assets/a4b42a64-c6db-4369-ac30-4dc397783369)

![Image](https://github.com/user-attachments/assets/4df9455f-3f20-4f13-a800-2e253fd18686)

**Comment by zhouak47** [comment]
It seems that we can have a meeting online by zoom or tencent meeting

**Comment by LJY-XCX** [comment]
Please share your availability and meeting link

**Comment by zhouak47** [comment]
周垒 邀请您参加腾讯会议
会议主题：周垒预定的会议
会议时间：2025/05/08 23:30-2025/05/09 01:00 (GMT+08:00) 中国标准时间 - 北京

点击链接入会，或添加至会议列表：
https://meeting.tencent.com/dm/07eh1RCArtNq

#腾讯会议：900-434-700

复制该信息，打开手机腾讯会议即可参与

**Comment by LJY-XCX** [comment]
Let's begin at 12am, Beijing Time

**Comment by zhouak47** [comment]
Ok.thanks a lot for your kind reply .This is the meeting link ,I am looking forward for your attandance
周垒 邀请您参加腾讯会议
会议主题：周垒预定的会议
会议时间：2025/05/09 12:00-13:30 (GMT+08:00) 中国标准时间 - 北京

点击链接入会，或添加至会议列表：
https://meeting.tencent.com/dm/hvgUG2pGkQDN

#腾讯会议：621-908-463

复制该信息，打开手机腾讯会议即可参与
